unit_testing/unitTests.py

```diff
@@ -10,8 +10,9 @@
     parser.add_argument('-v', '--verbose', action="store_true", help='Maximum output')
     args = parser.parse_args()
 
-    subprocess.run("cmake -S . -B build", shell=True)
-    subprocess.run("cmake --build build", shell=True)
+    dir=f"{args.regex}"
```

Comment:
As discussed:
* Separate directory and regex concepts (-R and -d args perhaps)
* If no directory argument, scan subfolders and execute each one (but not the build folder)

Reply:
This is because Regex is useful for running a single test out of a module, as well as selecting the module out from many modules

Reply:
- Separate directory and regex concepts using -D and -R are now implemented.

- If no directory argument is passed the then the unitTests.py scans the immediate subfolders (hence excluding nested build folders) and execute each one.